clinica/pipelines/dwi_preprocessing_using_t1/dwi_preprocessing_using_t1_utils.py

```diff
@@ -308,3 +308,34 @@ def prepare_reference_b0(in_dwi, in_bval, in_bvec, low_bval=5, working_directory
     )
 
     return out_reference_b0, out_b0_dwi_merge, out_updated_bval, out_updated_bvec
+
+
+def delete_temp_dirs(marker: str, dir_to_del: list, base_dir: str, delete_cache: bool):
+    """This function deletes the directory of the given list, if the pipeline is in "light_mode".
+
+    Parameters
+    ----------
+    marker: str
```

Comment:
Could we use a better name for this parameter? I found it hard to understand what it is from the docstring.

If I understand correctly, it is a path to some file, but we don't really care about the file itself, we just want to extract some part of it to filter the files to be deleted (subject_session ID maybe??).

I would first be clear on what you want to extract and give it an indicative name, then create a specific function

```python
def extract_whatever_you_want_to_extract(file_path: str) -> str:
    """Docstring with examples of extractions..."""
    return (Path(Path(file_path).parent).parent).name
```

and call it when connecting the delete node in the pipeline:

```python
(merge, delete_warp_field_tmp, [(("merged_file", extract_whatever), "marker")]),
```

And then simplify this function. 

Does this make sense?

Reply:
So it is used for two things:
 - one: to check that we are deleting the files related to the right subject-session, since the path contained in `marker`, contains the randomly generated name of the the folder of the right subject-session.
 - two: to be sure that we are past the point the files generated are useful to the pipeline and that we can delete them. In that way it is a checkpoint. 
 
For these reasons I chose to name it `marker` as it marks the location of a folder, and a "timepoint" in the pipeline.
Then again, i completely agree that this naming is far from clear to the reader.

I does make sense, will do something somewhat like you are suggesting to improve clarity !

Reply:
@NicolasGensollen I have done some changes going your way. WDYT ?